src/openms/source/ANALYSIS/ID/AScore.cpp

```diff
@@ -58,33 +58,47 @@ namespace OpenMS
   {
   }
 
-  PeptideHit AScore::compute(PeptideHit & hit, RichPeakSpectrum & real_spectrum, double fmt, Int number_of_phospho_sites)
+  PeptideHit AScore::compute(PeptideHit & hit, RichPeakSpectrum & real_spectrum, double fragment_mass_tolerance, bool fragment_mass_unit_ppm)
   {
-    String without_phospho_str(hit.getSequence().toString());
-    size_t found = without_phospho_str.find("(Phospho)");
-    while (found != string::npos)
+    String with_phospho_string = hit.getSequence().toString();
+
+    // count number of phosphorilation events
+    Size number_of_phosphorilation_events = 0;
+    for (Size i = with_phospho_string.find("Phospho"); i != std::string::npos; i = with_phospho_string.find("Phospho", i + 7))
     {
-      without_phospho_str.erase(found, String("(Phospho)").size());
-      found = without_phospho_str.find("(Phospho)");
+      ++number_of_phosphorilation_events;
     }
+
+    // create variant of the peptide with all phosphorilations removed
+    String without_phospho_str(hit.getSequence().toString());
+    without_phospho_str.substitute("(Phospho)", "");
     AASequence without_phospho = AASequence::fromString(without_phospho_str);
```

Comment:
If speed is a concern, this looks pretty inefficient.
How about going through the residues of the AASeq and removing "Phospho" if it shows up?
Should be about as fast as a .toString() and would also replace the couting loop above.